### PR TITLE
Adds option to allow prefixing the class name in an xunit report

### LINKED ISSUE #402: Allow user-specified prefix for all test names in XML report
I have a situation where a project contains several independent sub-packages, each with their own test suites.  To test the entire project, we run nosetests on each package separately from the package's top directory.  The resulting Xunit xml files are merged together and displayed within Hudson.

Unfortunately, since this is being done within each project's directory, the individual test class names don't include the package name, so the results get mangled and merged within Hudson.  A simple solution is to have the Xunit plugin support a "--xunit-prefix" option that gets prepended to the classname in the XML report.

The following changeset implements this option (and the unit tests for it):

https://code.google.com/r/jsiirola-python-nose-clone/source/detail?r=9c9026ef31cf7247391fff1650770cb93209a53f

Google Code Info:
Issue #: 390
Author: jsiir...@gmail.com
Created On: 2011-01-07T21:57:22.000Z
Closed On:

### PR DESCRIPTION
**Pull request by joscha**
In a matrixed build environment (for example running tests with `tox`) and producing multiple xunit reports, a report crawler (for example from Bamboo) can not distinguish between results in different environments, because the class name is always the same.
In order to allow distinguishing between failures in different environments, this PR adds an option `--xunit-prefix-with-testsuite-name` to the `nose-xunit` plugin.

A test result looking like this with `--with-xunit`:
```xml
<testcase classname="tests.x.y" name="somename" time="0.000">
```
becomes:
```xml
<testcase classname="nosetests.tests.x.y" name="somename" time="0.000">
```
with `--with-xunit --xunit-prefix-with-testsuite-name`
or:
```xml
<testcase classname="foo.tests.x.y" name="somename" time="0.000">
```
with `--with-xunit --xunit-testsuite-name=foo --xunit-prefix-with-testsuite-name`.

Side note: It works in a similar manner as the `useBrowserName` option in the [karma-junit-reporter](https://github.com/karma-runner/karma-junit-reporter).

Fixes #402

### CODE REVIEW

**Comment by dirk-thomas** [comment]
This patch implement a very valuable and long requested feature (#402).

I have the same problem that Jenkins only shows the `classname` but with multiple packages being tested it is unclear to which each of them belongs. The job already uses `--xunit-testsuite-name=...` but Jenkins does not utilize the `testsuite` `name` attribute. This new option would solve that.

File: unit_tests/test_xunit.py [hunk 1]
```diff
@@ -101,6 +113,39 @@ class DummyStream:
         f.close()
         return data
 
+class TestXMLOutputWithXMLAndPrefix(BaseTestXMLOutputWithXML):
+    def setUp(self):
+        super(TestXMLOutputWithXMLAndPrefix, self).setUp()
+        self.configure([
+            "--with-xunit",
+            "--xunit-file=%s" % self.xmlfile,
+            "--xunit-prefix-with-testsuite-name"
```
Comment (jszakmeister) [inline unit_tests/test_xunit.py]: This all looks good with one exception: I expected to see a test that uses xunit-testsuite-name to change the prefix.  Do you mind adding such a test?

Comment (joscha) [inline unit_tests/test_xunit.py]: I thought adding the default would be enough. Additional test is in 0214d2e66d0ef69dc36d3703f377ff8d0fc99f08

**Comment by joscha** [comment]
@jszakmeister I added the additional test in ab198aa and rebased to master

**Comment by jszakmeister** [comment]
Great!  Thanks!

**Comment by joscha** [comment]
Thanks for integrating!

**Comment by dirk-thomas** [comment]
Great! Can I assume that this will be released in 1.3.8 at some point in the future?

**Comment by jszakmeister** [comment]
Hopefully.  I'm not making any promises--I'm stepping down as maintainer soon.

**Pull request merged**